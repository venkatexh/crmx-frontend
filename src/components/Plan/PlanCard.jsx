import '../../sass/components/plan/planCard.component.scss'
import {saveSelectedPlan} from "../../redux/actions/plan/saveSelectedPlan";
import {useDispatch} from "react-redux";
import {useHistory} from "react-router";

const PlanCard = ({name, price, rate, action, description, features}) => {
  const dispatch = useDispatch();
  const history = useHistory();
  const handleActionClick = () => {
    dispatch(saveSelectedPlan({
      name,
      price,
      description
    }));
    history.push('/checkout');
  }

  return (
    <div className={'planCard'}>
      <div className={'planName'}>{name}</div>
      <div>
        <div className={'planPrice'}>{price}</div>
        <div className={'planRate'}>{rate}</div>
      </div>
      <div className={'planFeatures'}>{features}</div>
      <div>{description}</div>
      <div>
        <button className={`${action === 'Upgrade' ? 'upgradeButton' : 'currentButton'}`}
                onClick={handleActionClick}>{action}</button>
      </div>
    </div>
  )
}

export default PlanCard;