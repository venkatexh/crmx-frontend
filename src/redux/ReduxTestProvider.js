import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
