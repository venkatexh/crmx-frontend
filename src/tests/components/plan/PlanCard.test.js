import { shallow } from "enzyme";
import React from "react";
import PlanCard from "../../../components/Plan/PlanCard";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("Plan card component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <PlanCard
          name={"test card"}
          userPlan={"test plan"}
          description={"test description"}
          price={"99"}
          rate={"per month"}
          idx={"1234test"}
          selected={"1234test"}
          features={"this is a feature"}
        />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
