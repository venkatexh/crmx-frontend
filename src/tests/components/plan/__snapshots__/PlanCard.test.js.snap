// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan card component snapshot 1`] = `ShallowWrapper {}`;
